### ISSUE TITLE
SSR support

### ISSUE THREAD

**Issue by nataliawww** [AlbanCrepel/vue-displacement-slideshow#10]
Would be great to add SSR support so it works with nuxt.js too

**Comment by NikitaLoshchenin** [comment]
> Would be great to add SSR support so it works with nuxt.js too
Do you have any problems with production mode?
`i.setValue is not a function`  in browser dev tols with default nuxt configuration.config.js

**Comment by AlbanCrepel** [comment]
Hi @nataliawww and @NikitaLoshchenin , I am sorry I didn't have the opportunity to try nuxt.js yet... But when I do I'll have a look at your issue 👍

**Comment by nataliawww** [comment]
> > Would be great to add SSR support so it works with nuxt.js too
> > Do you have any problems with production mode?
> > `i.setValue is not a function`  in browser dev tols with default nuxt configuration.config.js

Yes, I have the same problem

**Comment by SixBytesUnder** [comment]
I tried using it as a client side plugin in nuxt.js
```
plugins: [
	{ src: '~/plugins/vue-displacement-slideshow', mode: 'client' }
	// { src: '~/plugins/vue-displacement-slideshow', ssr: false } // pre nuxt 2.4.0
],
```
but it fails with error message:
`[Vue warn]: Error in beforeCreate hook: "ReferenceError: document is not defined"`

@AlbanCrepel please look into integrating with nuxt :)

**Comment by AlbanCrepel** [comment]
Hi @SixBytesUnder ! I just started using Nuxt so I'll dig into this issue as soon as I can !

**Comment by SixBytesUnder** [comment]
Fantastic, thank you :)
If you're interested, I want to use your plugin to nicely change background images on my project here https://github.com/SixBytesUnder/infoboard

**Comment by AlbanCrepel** [comment]
Hi everyone ! 
@SixBytesUnder This syntax can't work :
> plugins: [
	{ src: '~/plugins/vue-displacement-slideshow', mode: 'client' }
	// { src: '~/plugins/vue-displacement-slideshow', ssr: false } // pre nuxt 2.4.0
],

Because vue-displacement-slideshow is not a plugin, it's just a component.
So if you want to use it with nuxt, you can wrap it between `<no-ssr></no-ssr>` tags, as explained in the documentation [here](https://nuxtjs.org/api/components-no-ssr/). It could look like this : 

```html
<no-ssr>
    <vue-displacement-slideshow />
</no-ssr>
```

I tried it and it works, both in development and in production. I hope this solution is suitable for you all @nataliawww @NikitaLoshchenin @SixBytesUnder !

**Comment by andreasantonsson** [comment]
I'm running a universal nuxt.js (2.7.1) app. The `<no-ssr>` did the trick in development, but I'm still getting the `i.setValue is not a function` in production. 

Anyone else having the same issue still?

**Comment by nla1989** [comment]
All setValue methods in the map are undefined in production mode with universal mode in Nuxt.js.
![image](https://user-images.githubusercontent.com/49455662/59707307-70710600-9213-11e9-8f8e-f33cf60cf13c.png)
@AlbanCrepel Do you have any idea how to fix this error? I will be very grateful for any tip, as long as I have no idea what could be the problem. )

**Comment by AlbanCrepel** [comment]
@nla1989 I know about the problem, and have already tried to have a look at it and fix it, without success. When a copy paste the component and use it in a nuxt project, it works. But when I use the library with the built files, it doesn't work because of the error you mentioned. I will try my best to fix this 🙏 
And for your information, I also worked on a version implementing the displacement effect on videos ! But this will take time to deploy...

**Comment by AlbanCrepel** [comment]
Hi @nla1989 @andreasantonsson @SixBytesUnder @nataliawww @NikitaLoshchenin !
I am glad to announce that the bug is fixed in the last release of this component (v1.2.0) ! :tada: 
It is now working in Nuxt projects in production mode. The problem came from the minification of build files from webpack...
